### PR TITLE
New packages: qt6

### PR DESCRIPTION
**Pull request by Johnnynator**
TOOD:
- [ ] Address TODOs

~~the cmake part does not use the host version of some tools e.g. a problem with lconvert~~ fixed with 6.1.0

[ci skip]

### CODE REVIEW

**Comment by ericonr** [comment]
Missing libatomic from armv6l. Do you know why armv6l doesn't get this error, only armv6l-musl?

**Comment by Johnnynator** [comment]
> Missing libatomic from armv6l. Do you know why armv6l doesn't get this error, only armv6l-musl?

armv6l does error, you just didn't see it, since CI doesn't run for armv6l

File: srcpkgs/qt6-base/template [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+ sqlite-devel Vulkan-Headers mit-krb5-devel vulkan-loader"
+short_desc="Cross-platform application and UI framework (QT6)"
+maintainer="John <me@johnnynator.dev>"
+license="GPL-3.0-only with Qt-GPL-exception-1.0, LGPL-3.0-only, GPL-2.0-or-later"
+homepage="https://www.qt.io"
+distfiles="https://download.qt.io/development_releases/qt/${version%.*}/${version/rc/-rc}/submodules/qtbase-everywhere-src-${version/rc/-rc}.tar.xz"
+checksum=3884c64750a9b6ef544ee2fb72a2bf5e5e167501d6ea11ab1a55105e354a506e
+python_version=3
+
+# TODO: fix usage of ccache
```
Comment (yopito) [inline srcpkgs/qt6-base/template:32]: I had a look on qt6-base and cmake. here my points:

"`-D-DQT_USE_CCACHE`" has no effect if ccache is enabled from xbps-src  : ccache will be used.
I think "`-DQT_USE_CCACHE=OFF`" might be a better idea (similar to other packages)

qt6-base build uses precompiled headers (pch) when working with automoc. ccache does not handle pch until some additional configuration (see https://ccache.dev/manual/4.2.1.html#_precompiled_headers).
 => so ccache is rather useless in this case: about 150/1160 "objects" are stored in ccache
workarounds:
* hope that packagers will enable pch support in ccache. IMHO, this support is somewhat imprecise, afraid of false positive
* disable use of pch for building qt6-base with "`-DBUILD_WITH_PCH=OFF`" : some overhead (not that much) on first build, but ccache efficiency after (about 70 minutes on 1st run, 6 minutes  then to build+package qt6-base). And no particular settings

File: common/build-style/cmake.sh [hunk 2]
```diff
@@ -49,7 +49,7 @@ SET(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)
 SET(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)
 SET(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)
 _EOF
-		cmake_args+=" -DCMAKE_TOOLCHAIN_FILE=cross_${XBPS_CROSS_TRIPLET}.cmake"
+		cmake_args+=" -DCMAKE_TOOLCHAIN_FILE=${wrksrc}/${cmake_builddir}/cross_${XBPS_CROSS_TRIPLET}.cmake"
```
Comment (yopito) [inline common/build-style/cmake.sh:52]:
```suggestion
		cmake_args+=" -DCMAKE_TOOLCHAIN_FILE=${wrksrc}/${build_wrksrc}/${cmake_builddir}/cross_${XBPS_CROSS_TRIPLET}.cmake"
```

**Pull request merged**